### ISSUE TITLE
[Feature Request] Support custom CSS themes

### ISSUE THREAD

**Issue by subdavis** [mickael-kerjean/filestash#331]
Preferably via runtime configuration through the console.   I'd like to be able to change some of the look and feel of filestash to match another application. Nothing major, just colors and maybe adding some border radius.

Related: 

#246 -- I don't think this issue should have been closed.  This was not a support request, and there's no documentation on this. 
#323 -- This PR isn't linked to an issue.  There should be an issue to discuss how this is implemented.

**Comment by mickael-kerjean** [comment]
Thank you for opening this, such a feature would be nice to have indeed

> #246 -- I don't think this issue should have been closed. This was not a support request, and there's no documentation on this.

My mistake, I only read the first word with the `support` tag when the issue template says: `Please, don't create tickets on github for support. Instead, you can either: ...` and stopped reading there.

> #323 -- This PR isn't linked to an issue. There should be an issue to discuss how this is implemented.

Definitely. The back story is I really want Filestash to become one day a fat binary one can download and start directly like any other CLI tool. There's still a long way to go but at this stage I'm awaiting for [this proposal](https://go.googlesource.com/proposal/+/master/design/draft-embed.md) to move forward.

Ultimately the entire public directory will eventually disappear to be embedded in the binary instead. An approach that would make this feature possible considering the long term plan would be to create a custom field within the admin console and have the index.html or another CSS file to load the custom stylesheet that will be loaded directly from a backend endpoint that will simply dump the content of the new CSS field

**Comment by thielepaul** [comment]
> Definitely. The back story is I really want Filestash to become one day a fat binary one can download and start directly like any other CLI tool. There's still a long way to go but at this stage I'm awaiting for this proposal to move forward.

I just created a new PR (#332) implementing your proposal  @mickael-kerjean

**Comment by mickael-kerjean** [comment]
merged! Thank you very much for your contribution :)

**Comment by subdavis** [comment]
Awesome!   I'll let @mickael-kerjean close this issue.  This solution meets my needs. Thank you.

**Comment by subdavis** [comment]
Just wanted to share this as sort of a "thankful user" and maybe share a use case you hadn't anticipated.

I'm working on a data management system over at https://github.com/subdavis/workspaces-io built on top of s3, minio, and filestash.  It's basically an indexing and role-based-access-control layer on top of these systems.  

* I wrote an implementation of the filesystem interface that basically handles permissions checking then forwards to s3
* I wrapped the client with some additional functionality (screenshots below) and adapted filestash to my look-and-feel.

![Screenshot from 2020-10-18 16-01-39](https://user-images.githubusercontent.com/4214172/96378745-1ad8a080-115c-11eb-9f7d-18aef5d80b4e.png)
![Screenshot from 2020-10-18 16-01-55](https://user-images.githubusercontent.com/4214172/96378747-1b713700-115c-11eb-9e88-2a2cb1f85ca8.png)

I'm really happy with this solution, and I may have some additional asks (I'll implement myself) for better 

* I'd like to expose session cookies through `params` to the filesystem interface for transparent authentication.  This will work if I have filestash and my wrapping app serving from the same origin
* I'd like to implement a lightweight event system inside the filestash client to notify listeners of route changes using https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage

I'll make future issues to talk more about these things once I've had a chance to play with them a bit. 

Mostly just wanted to say that this is a great product with a lot of flexibility, and you probably didn't expect to see a use like this :)
